tools/build_langserver/langserver/analyzer.go

```diff
@@ -155,6 +167,101 @@ func (a *Analyzer) IdentFromFile(uri lsp.DocumentURI) ([]*Identifier, error) {
 	return idents, nil
 }
 
+// BuildLabelFromString returns a BuildLabel object,
+func (a *Analyzer) BuildLabelFromString(ctx context.Context, rootPath string,
+	uri lsp.DocumentURI, labelStr string) (*BuildLabel, error) {
+	filepath, err := GetPathFromURL(uri, "file")
+	if err != nil {
+		return nil, err
+	}
+
+	label, err := core.TryParseBuildLabel(labelStr, filepath)
+	if err != nil {
+		return nil, err
+	}
+	if label.IsEmpty() {
+		return nil, fmt.Errorf("invalid build label %s", labelStr)
+	}
+
+	// Get the BUILD file path for the build label
+	var labelPath string
+	// Handling subrepo
+	if label.Subrepo != "" {
+		return &BuildLabel{
+			BuildLabel:      &label,
+			Path:            label.PackageDir(),
+			BuildDef:        nil,
+			BuildDefContent: "Subrepo label: " + labelStr,
+		}, nil
+	}
+
+	if label.PackageName == filepath {
```

Comment:
Now this is tricky, if the BUILD file is configurable.

So basically the reason why I didn't concate the BUILD file name in the end is because I ran into the case when I pass in a relative buildtarget, like ":langserver", what happens is, the BuildLabel object I get would have a `packageDir` as the buildfile itself instead of the directory. So I did this to avoid this type of conflict.